packages/core/test/integration/query-interface.test.js

```diff
@@ -256,23 +263,23 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
         }, {
           schema: 'schema',
         });
-        await this.queryInterface.addIndex(
-          { schema: 'schema', tableName: 'table' },
-          ['name', 'isAdmin'],
-          null,
-          'schema_table',
-        );
-        const indexes = await this.queryInterface.showIndex({
+        await this.queryInterface.addIndex({ schema: 'schema', tableName: 'table' }, { fields: ['name', 'isAdmin'] });
+        const indexes = await this.queryInterface.showIndexes({
           schema: 'schema',
           tableName: 'table',
         });
         expect(indexes.length).to.eq(1);
-        expect(indexes[0].name).to.eq('table_name_is_admin');
+        // MariaDB & MySQL support index via alter and if no name is given, the db sets the name not sequlize
+        if (['mariadb', 'mysql'].includes(dialectName)) {
+          expect(indexes[0].name).to.eq('name');
```

Comment:
we should not let the database name the index, this index should be named `table_name_is_admin` no matter the database, otherwise we won't be able to know what the index is named in the database and sync won't be able to match the model definition with the database

Reply:
I've updated the unit test and included an integration test to test this.